tests/integration/cmd_devfile_init_test.go

```diff
@@ -377,30 +381,73 @@ var _ = Describe("odo devfile init command tests", func() {
+					// An issue from the Devfile library causes the parent to not be resolved properly:
+					//  ✗  unable to parse devfile: failed to populateAndParseDevfile: url path to directory or file should contain 'tree' or 'blob'.
+					//
+					// Even after using a URL with 'tree'or 'blob', the parser still does not pass.
+					// This workaround downloads the parent manually and references it via the local filesystem.
+					srcDir := helper.CreateNewContext()
+					DeferCleanup(func() {
+						helper.DeleteDir(srcDir)
+					})
+					parentDevfile := filepath.Join(commonVar.Context, "parent-devfile.yaml")
+					f, err := os.Create(parentDevfile)
+					Expect(err).ShouldNot(HaveOccurred())
+					resp, err := http.Get("https://github.com/OpenLiberty/application-stack/releases/download/maven-0.7.0/devfile.yaml")
+					Expect(err).ShouldNot(HaveOccurred())
+					defer resp.Body.Close()
+					_, err = io.Copy(f, resp.Body)
+					Expect(err).ShouldNot(HaveOccurred())
+					childDevfile = filepath.Join(srcDir, "devfile-child.yaml")
+					helper.CopyExampleDevFile(filepath.Join("source", "devfiles", "nodejs", "devfile-with-parent.yaml"), childDevfile)
+					helper.ReplaceString(childDevfile, "https://github.com/OpenLiberty/application-stack/releases/download/maven-0.7.0/devfile.yaml", "parent-devfile.yaml")
+
+					b, err := os.ReadFile(childDevfile)
 					Expect(err).To(BeNil())
-					var content map[string]interface{}
-					Expect(yaml.Unmarshal(originalDevfileContent, &content)).To(BeNil())
-					for k := range content {
-						originalKeyList = append(originalKeyList, k)
-					}
+					originalDevfileContent = string(b)
+
+					helper.Cmd("odo", "init", "--name", "aname", "--devfile-path", childDevfile).ShouldPass()
 				})
 
-				It("should not replace the original devfile", func() {
-					helper.Cmd("odo", "init", "--name", "aname", "--devfile-path", srcDevfile).ShouldPass()
-					devfileContent, err := os.ReadFile(filepath.Join(commonVar.Context, "devfile.yaml"))
-					Expect(err).To(BeNil())
-					var content map[string]interface{}
-					Expect(yaml.Unmarshal(devfileContent, &content)).To(BeNil())
-					for k := range content {
-						Expect(k).To(BeElementOf(originalKeyList))
-					}
+				It("should create an effective Devfile", func() {
+					devfilePath := filepath.Join(commonVar.Context, "devfile.yaml")
+
+					By("not replacing the original Devfile", func() {
+						// Re-read the original Devfile
+						originalB, err := os.ReadFile(childDevfile)
+						Expect(err).To(BeNil())
+						originalDevfileContentReread := string(originalB)
+						diff := cmp.Diff(originalDevfileContentReread, originalDevfileContent)
+						Expect(diff).Should(BeEmpty())
+					})
+
+					By("creating a different Devfile", func() {
+						b, err := os.ReadFile(filepath.Join(commonVar.Context, "devfile.yaml"))
+						Expect(err).To(BeNil())
+						devfileContent := string(b)
+						Expect(devfileContent).ShouldNot(BeEquivalentTo(originalDevfileContent))
+					})
```

Comment:
Is it what we want? We are changing the behaviour, as `odo init` was copying the child devfile as is before

Reply:
Whoops - I missed your comments..
Indeed, previously, the child Devfile was copied as is. Then it was parsed by `odo init` to set user-defined fields like the component `name`. The test here was previously asserting that the resulting Devfile had the same fields as the original one.
But now, with the new changes in the Devfile library, this Devfile can no longer be parsed. My initial goal was to make `odo init` always return an effective Devfile as a result. But with your comments, I agree this is a change in behavior and can be discussed and considered separately.
I'll revert the changes done here and report the parsing issue to the Devfile library.

Reply:
Issue reported in https://github.com/devfile/api/issues/1119, and changes done in [`3603642` (#6768)](https://github.com/redhat-developer/odo/pull/6768/commits/3603642d6e8f739ccdceafa6ba6770f5e602170c)